src/compiler/checker.ts

```diff
@@ -21951,7 +21956,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     // reveal the reason).
                     // We can switch on `reportErrors` here, since varianceCheckFailed guarantees we return `False`,

```

Comment:
This mentions the code I deleted, so I'd have to delete this if this fix is right.